sdk/servicebus/service-bus/src/serviceBusMessage.ts

```diff
@@ -1065,21 +1033,13 @@ export class ServiceBusMessage implements ReceivedMessage {
       );
     } else if (this.delivery.remote_settled) {
       error = new Error(`Failed to ${operation} the message as this message is already settled.`);
-    } else if (!receiver || !receiver.isOpen()) {
-      const errorMessage =
-        `Failed to ${operation} the message as the AMQP link with which the message was ` +
-        `received is no longer alive.`;
-      if (this.sessionId != undefined) {
-        error = translate({
-          description: errorMessage,
-          condition: ErrorNameConditionMapper.SessionLockLostError
-        });
-      } else {
-        error = translate({
-          description: errorMessage,
-          condition: ErrorNameConditionMapper.MessageLockLostError
-        });
-      }
+    } else if ((!receiver || !receiver.isOpen()) && this.sessionId !== undefined) {
+      error = translate({
+        description:
+          `Failed to ${operation} the message as the AMQP link with which the message was ` +
+          `received is no longer alive.`,
+        condition: ErrorNameConditionMapper.SessionLockLostError
```

Comment:
You need an early exit from `throwIfMessageCannotBeSettled()` if the message is a deferred one. Otherwise, you will end up throwing this error where as we can very well settle the message using the management link.

Either make an early exit or not call `throwIfMessageCannotBeSettled()` at all for deferred messages.
Before this PR, I believe `throwIfMessageCannotBeSettled()` was never invoked for deferred messages